st3/sublime_lib/region_manager.py

```diff
@@ -0,0 +1,61 @@
+import sublime
+
+from ._compat.typing import Optional, List, TypeVar
+from .flags import RegionOption
+
+
+__all__ = ['RegionManager']
+
+
+T = TypeVar('T')
+
+
+def coalesce(*values: Optional[T]) -> T:
+    return next(value for value in values if value is not None)
+
+
+class RegionManager:
+    def __init__(
+        self,
+        view: sublime.View,
+        key: Optional[str] = None,
+        *,
+        scope: Optional[str] = None,
+        icon: Optional[str] = None,
+        flags: Optional[RegionOption] = None
+    ):
+        self.view = view
+
+        if key is None:
+            self.key = str(id(self))
+        else:
+            self.key = key
+
+        self.scope = scope
+        self.icon = icon
+        self.flags = flags
+
+    def __del__(self) -> None:
+        self.erase()
+
+    def set(
+        self,
+        regions: List[sublime.Region],
```

Comment:
The implementation in `sublime` requires annotations to be a list, but regions seemingly only need to have a length and be iterable, so a tuple also works (sets and dicts don't because `Region`s aren't hashable). I think the closest abstract match is a `Collection`.

Reply:
Do you think we should change the `sublime` type stub?

Reply:
Yeah.

Reply:
Done.